src/org/labkey/test/pages/study/ManageDatasetQCStatesPage.java

```diff
@@ -0,0 +1,96 @@
+{
+    public ManageDatasetQCStatesPage(WebDriver driver)
+    {
+        super(driver);
+    }
+
+    public static ManageDatasetQCStatesPage beginAt(WebDriverWrapper driver)
+    {
+        return beginAt(driver, driver.getCurrentContainerPath());
+    }
+
+    public static ManageDatasetQCStatesPage beginAt(WebDriverWrapper driver, String containerPath)
+    {
+        driver.beginAt(WebTestHelper.buildURL("controller", containerPath, "action"));
+        return new ManageDatasetQCStatesPage(driver.getDriver());
+    }
+
+    @Override
+    protected void waitForPage()
+    {
+        waitForText("Manage Dataset QC States");
+    }
+
+    public ManageDatasetQCStatesPage addStateRow(String state, String description, boolean publicData)
+    {
+        Locator.tagWithClass("span", "fa-plus-circle").waitForElement(elementCache().stateForm, 1000)
+                .click();               // there is only ever on
+        Locator lastLabelLoc = Locator.xpath("(//input[@name='newLabels'])[last()]");
+        Locator lastDescLoc = Locator.xpath("(//input[@name='newDescriptions'])[last()]");
+        Locator lastCheckbox = Locator.xpath("(//input[@type='checkbox' and @name='newPublicData'])[last()]");
+
+        setFormElement(lastLabelLoc, state);
+        setFormElement(lastDescLoc, description);
+        new Checkbox(lastCheckbox.findElement(elementCache().stateForm)).set(publicData);
+
+        return this;
+    }
+
+    public ManageDatasetQCStatesPage setStatePublic(String id, boolean publicData)
+    {
+        new Checkbox(Locator.checkboxById(id).findElement(elementCache().stateForm)).set(publicData);
+        return this;
+    }
+
+    public ManageDatasetQCStatesPage setDefaultAssayQCState(String state)
+    {
+        selectOptionByText(Locator.name("defaultAssayQCState"), state);
+        return this;
+    }
+
+    public ManageDatasetQCStatesPage setDefaultDirectEntryQCState(String state)
+    {
+        selectOptionByText(Locator.name("defaultDirectEntryQCState"), state);
+        return this;
+    }
+
+    public ManageDatasetQCStatesPage setDefaultPipelineQCState(String state)
+    {
+        selectOptionByText(Locator.name("defaultPipelineQCState"), state);
+        return this;
+    }
+
+    public ManageDatasetQCStatesPage showPrivateDataByDefault(String state)
+    {
+        selectOptionByText(Locator.name("showPrivateDataByDefault"), state);
+        return this;
+    }
+
+    public ManageStudyPage clickSave()
+    {
+        clickButton("Save");
+        return new ManageStudyPage(getDriver());
+    }
+
+    protected ElementCache newElementCache()
+    {
+        return new ElementCache();
+    }
+
+    protected class ElementCache extends LabKeyPage.ElementCache
+    {
+        WebElement stateForm = Locator.tagWithAttribute("form", "name", "manageQCStates")
+                .findWhenNeeded(this).withTimeout(WAIT_FOR_JAVASCRIPT);
```

Comment:
The `select`s for the various defaults should be in the element cache ('defaultAssayQCState', 'defaultDirectEntryQCState', 'defaultPipelineQCState', and 'showPrivateDataByDefault')